### PR TITLE
fix: [CDS-35958]: Infra tab completion tick and code duplication removal

### PR DESCRIPTION
**Pull request by satyam-harness**
##### Jira Links:
https://harness.atlassian.net/browse/CDS-35958

##### Screenshots:

before
Screenshot 2022-04-05 at 12 30 31 PM

after
Screenshot 2022-04-05 at 12 30 09 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/Serverless/ServerlessStep.module.scss [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/*
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/Serverless/ServerlessStep.module.scss:1]: Keep this file name same as component name

File: src/modules/75-cd/components/PipelineSteps/ServerlessAzure/__test__/__snapshots__/ServerlessAzureSpec.test.tsx.snap [hunk 2]
```diff
@@ -20,7 +20,7 @@ exports[`Test ServerlessAzureSpec snapshot should render edit view for inputset
           >
             <span
               class="Tooltip--acenter bp3-label"
-              data-tooltip-id="serverlessAzureConnector"
+              data-tooltip-id="awsInfraConnector"
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/ServerlessAzure/__test__/__snapshots__/ServerlessAzureSpec.test.tsx.snap:23]: If we need to have different tooltip for different types of connector then these changes are wrong.

File: src/modules/75-cd/components/PipelineSteps/ServerlessGCP/__test__/__snapshots__/ServerlessGCPSpec.test.tsx.snap [hunk 3]
```diff
@@ -20,7 +20,7 @@ exports[`Test ServerlessGCPSpec snapshot should render edit view for inputset vi
           >
             <span
               class="Tooltip--acenter bp3-label"
-              data-tooltip-id="serverlessGCPInfraConnector"
+              data-tooltip-id="awsInfraConnector"
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/ServerlessGCP/__test__/__snapshots__/ServerlessGCPSpec.test.tsx.snap:23]: If we need to have different tooltip for different types of connector then these changes are wrong.

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/PipelineSteps/Serverless/ServerlessSpecEditable.tsx [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+    </Layout.Vertical>
+  )
+}
+
+export const ServerlessVariablesForm: React.FC<ServerlessSpecEditableProps> = ({
+  metadataMap,
+  variablesData,
+  initialValues
+}) => {
+  const infraVariables = variablesData?.infrastructureDefinition?.spec
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/PipelineSteps/Serverless/ServerlessSpecEditable.tsx:303]: variablesData is a mandatory prop from ServerlessSpecEditableProps interface --> Please check if we need to use optional chaning while reading infrastructureDefinition from it

File: src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx [hunk 5]
```diff
@@ -365,6 +380,13 @@ export default function DeployInfraSpecifications(props: React.PropsWithChildren
                 'ServerlessAzureFunctions'
               )
             }
+            customStepProps={{
+              formInfo: {
+                formName: 'serverlessAzureInfra',
+                type: 'Gcp',
+                header: getString('pipelineSteps.awsConnectorLabel')
```
[the remark this answers is not in the document]
Comment (satyam-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx]: that will be changed in next pr, type 'azure' is not added in services. (currently we have disabled gcp and azure both).

[the remark this answers is not in the document]
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx]: Let's fix this as well as you are already fixing other comments

**Comment by satyam-harness** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineSteps/Serverless/ServerlessSpecEditable.tsx (hunk 4 above)
Comment (satyam-harness) [reply src/modules/75-cd/components/PipelineSteps/Serverless/ServerlessSpecEditable.tsx]: this was earlier added for testcases

**Comment by satyam-harness** [comment]
retrigger jest

File: src/modules/70-pipeline/utils/stageHelpers.ts [hunk 6]
```diff
@@ -213,3 +213,51 @@ export const getSelectedDeploymentType = (
...
+          }
+        }
+      }
+    case ServiceDeploymentType.ServerlessAzureFunctions:
+      return {
+        formInfo: {
+          formName: 'serverlessAzureInfra',
+          // @TODO - (change type to 'azure')
+          // this is not being used anywhere currently, once azure support is there we will change it.
+          type: 'Gcp',
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/utils/stageHelpers.ts:238]: Instead of hardcoding, can we have enum/maps for the types

**Review by chetanvekariya-harness — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineSteps/ServerlessInfraSpec/ServerlessInfraSpec.module.scss [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+.connectorRef {
+  padding-bottom: 0px !important;
+}
+
+.accordionPanel {
+  padding-top: var(--spacing-small) !important;
+}
+
+.accordionDetails {
+  padding-bottom: 0px !important;
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/ServerlessInfraSpec/ServerlessInfraSpec.module.scss:43]: `!important` is used at a lot of places. Hope all are intentional.

File: src/modules/70-pipeline/utils/stageHelpers.ts (hunk 6 above)
Comment (chetanvekariya-harness) [reply src/modules/70-pipeline/utils/stageHelpers.ts:238]: Fixed.

**Comment by chetanvekariya-harness** [comment]
retrigger jest

**Comment by chetanvekariya-harness** [comment]
retrigger jest

**Review by chetanvekariya-harness — APPROVED** [review]

**Comment by chetanvekariya-harness** [comment]
retrigger jest

**Comment by chetanvekariya-harness** [comment]
trigger cypress

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**